### ISSUE TITLE
Create a pytest plugin to better show progress in execution of tests

### ISSUE THREAD

**Issue by cairomassimo** [turingarena/turingarena#22]
In our use case, the tests are usually low in number but each takes a long time to run.
For this reason, it would be nice to have a graphical view showing:

- which tests are currently running (with `pytest-xdist` there can be more running in parallel)
- how long each test is taking
- a live view of the logs

which can be implemented as a pytest plugin replacing `terminalreporter` (see for example https://github.com/Frozenball/pytest-sugar)
This plugin could take the whole terminal while the tests are running (like an ncurses app). Then, when the test finish or are interrupted, it could show a standard report (i.e., based on the original code of pytest) on the terminal.
